lib/ex_doc/retriever.ex

```diff
@@ -235,18 +239,25 @@ defmodule ExDoc.Retriever do
        ) do
     {:docs_v1, _, _, content_type, _, module_metadata, _} = module_data.docs
     {{type, name, arity}, anno, signature, source_doc, metadata} = doc_element
+    doc_file = anno_file(anno, source)
     doc_line = anno_line(anno)
-    source = anno_file(anno, source)
+
+    ## When do we not have a source_line?
```

Comment:
We should always have one, no? If we are not sure, remove it and we will know if it is ever an issue. :D

Reply:
The original seemed to think that it was possible for us to not have a line, so I assumed that there was a reason for that. I'm happy to remove it and see what breaks :)

Reply:
I guess the bigger question is: which one should win? We have both source_line and source_file and anno_file and anno_line. Shouldn't they be the same? When they are not the same? And we use doc_file and doc_line further down below as well. Shouldn't we use the same throughout?

The `Anno` in EEP 48 is meant to be the Anno of the source code, so shouldn't it always match the source_file+source_line?

Reply:
> The Anno in EEP 48 is meant to be the Anno of the source code, so shouldn't it always match the source_file+source_line?

EEP-48 Anno is the location of the documentation, not the source code. It is what we use to generate warnings for documentation issues, such as broken links etc.

We get the source code location from the AST.

Most of the time EEP-48 Anno does not contain any file, and then we assume that the docs are in the same file as the source code.

Reply:
Ok, so this all makes sense then. And I actually know when we won't have a source line. It is for Elixir's `Kernel.SpecialForm`, where we have stub definitions of functions/macros. However, this only applies to functions/macros. We will always have a line for modules, types, callbacks, and so on.

Given this is a Elixir specific issue, what if we solve this on the Elixir side things, and fallback to `anno_line` there, and in here we assume `source_line` always exists?

Reply:
yes, that make sense.